sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1470,33 +1470,27 @@ public final void run() {
             long currentNanoTime = System.nanoTime();
 
             while (true) {
-                try {
-                    AcquireListener removedTask = this.pool.pendingAcquisitions.poll();
-                    if (removedTask == null) {
-                        // queue is empty
-                        break;
-                    }
+                AcquireListener removedTask = this.pool.pendingAcquisitions.poll();
+                if (removedTask == null) {
+                    // queue is empty
+                    break;
+                }
 
-                    long expiryTime = removedTask.getAcquisitionTimeoutInNanos();
+                long expiryTime = removedTask.getAcquisitionTimeoutInNanos();
 
-                    // Compare nanoTime as described in the System.nanoTime documentation
-                    // See:
-                    // * https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#nanoTime()
-                    // * https://github.com/netty/netty/issues/3705
-                    if (expiryTime - currentNanoTime < 0) {
-                        this.onTimeout(removedTask);
-                    } else {
-                        if (!this.pool.pendingAcquisitions.offer(removedTask)) {
-                            logger.error("Unexpected failure when returning the removed task"
-                                    + " to pending acquisition queue. current size [{}]",
-                                this.pool.pendingAcquisitions.size());
-                        }
-                        break;
+                // Compare nanoTime as described in the System.nanoTime documentation
+                // See:
+                // * https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#nanoTime()
+                // * https://github.com/netty/netty/issues/3705
+                if (expiryTime - currentNanoTime < 0) {
+                    this.onTimeout(removedTask);
+                } else {
+                    if (!this.pool.pendingAcquisitions.offer(removedTask)) {
```

Comment:
I think my biggest question is: if we can not offer this task back, what would happen for this task? we did not fail it or retry, so any possibility it will stuck somewhere?

PriorityBlockingQueue will never return false for the offer method, then I guess it should be fine here?

Reply:
Yes - it is fine. the queue has unlimited capcity

Reply:
I mean limited only by resource constraints - and since admission control is in place there is no issue.

Reply:
got it, thanks for the explanation~